src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_CAPACITY = 1.5;
+    private static final int ARRAYS_ZERO_BORDER = 0;
+    private static final int INCREASE_INDEX_FOR_COPY = 1;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elementData.length == size) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (elementData.length == size) {
+            grow();
+        }
+        size++;
+        if (checkIndex(index)) {
```

Comment:
we can place `throw new ArrayListIndexOutOfBoundsException("Index out of bounds array size");` into `checkIndex` method, this will reduce number of duplicated code